source/schwarz_base.cpp

```diff
@@ -339,10 +365,31 @@ void SchwarzBase<ValueType, IndexType>::run(
     auto start_time = std::chrono::steady_clock::now();
     int num_converged_procs = 0;
 
+    //CHANGED - file writing
+    char send_name[30], recv_name[30], pe_str[3];
+    sprintf(pe_str, "%d", metadata.my_rank);
+
+    strcpy(send_name, "send");
+    strcat(send_name, pe_str);
+    strcat(send_name, ".txt");
+
+    strcpy(recv_name, "recv");
+    strcat(recv_name, pe_str);
+    strcat(recv_name, ".txt");
+
+    std::ofstream fps; //file for sending log
+    fps.open(send_name);
+
+    std::ofstream fpr; //file for receiving log
+    fpr.open(recv_name);
+    
+    if(metadata.my_rank == 0) std::cout << "Constant - " << metadata.constant << ", Gamma - " << metadata.gamma <<std::endl;
```

Comment:
Also remove these `std::cout`s, they completely skew the timings.

Reply:
this is outside the loop.

Reply:
Well, I had started the timer after all this printing and file opening. The recent merges somehow messed that!